src/pudl/extract/excel.py

```diff
@@ -167,24 +168,25 @@ def process_final_page(df, page):
         return df
 
     @staticmethod
-    def get_dtypes(year, page):
-        """Provide custom dtypes for given page and year."""
+    def get_dtypes(part, page):
+        """Provide custom dtypes for given page and partition."""
         return {}
 
-    def extract(self, years):
+    def extract(self, partitions):
```

Comment:
do you mean passing in kwargs into extract like `extract(year=1, year=2, year=3)` or passing in `extract(year=[1,2,3,])`? for the former, i'm not sure how you would distinguish between the years. and if it is the latter, I'm not sure how to convert `extract`'s kwargs into `get_resources`'s kwargs.

Reply:
You can't pass named argument more than once so it'd have to be `years=[1,2,3]`. My idea was to add helper function that will fan-out the multi-valued partition into sequence of single-valued assignments, e.g. `{month=9, year=[1,2]} --> [{month=9, year=1}, {month=9, year=2}]`.

Here is the helper function:
```
import itertools

def iterate_multivalue_dict(**kwargs):
    single_valued = {k: v for k, v in kwargs.items() if type(v) != list}

    # Transform multi-valued {k: vlist} into {k1: [{k1: v1}, {k1: v2}, ...], k2: [...], ...}
    multi_valued = {k: [{k: v} for v in vlist] for k, vlist in kwargs.items() if type(vlist) == list}

    for value_assignments in itertools.product(*multi_valued.values()):
        result = dict(single_valued)
        for k_v in value_assignments:
            result.update(k_v)
        yield result
```

Reply:
hey @rousik! I'm wondering if you can give me some additional pointers on this or poke at it yourself.

Even with this helper function, I can't seem to figure out how to pass **partitions like `year=[2009,2019]` into `pudl.extract.excel.GenericExtractor.extract` which then gets converted into a different set of kwargs for the `load_excel_file` method (or the rest of the methods which currently take `partition`. I'm assume this is simple and I'm just not getting it because I've really never mucked with kwargs.

Reply:
pause on that, I think I may have figured it out! lol.